docs/src/docs/asciidoc/user/thread-pools.adoc

```diff
@@ -0,0 +1,113 @@
+---
+---
+ifndef::sourcedir[]
+include::common.adoc[]
+endif::sourcedir[]
+
+= Thread Pools
+
+
+ifdef::notBuildingForSite[]
+include::menu.adoc[]
+endif::notBuildingForSite[]
+
+== Intro
+
+Some services work asynchronously, hence they require thread pools to perform their tasks. All thread pooling
+facilities are centralized behind the `ExecutionService` interface.
+
+== Let's start with a bit of theory
+
+=== What `ExecutionService` provides
+
+ * `ScheduledExecutorService` with which one can scheduled tasks, ie: tasks that happen repeatedly after a configurable
+   delay.
+ * Unordered `ExecutorService` to which tasks can be submitted and are going to be executed as soon as an available
+   thread is free.
+ * Ordered `ExecutorService` which provides a service comparable to the one above, with the extra guarantee that tasks
+   are going to be executed in the order they were submitted.
+
+=== Available `ExecutionService` implementations
+
+ * `OnDemandExecutionService` creates a new pool each time an executor service (scheduled or not) is requested.
+   This implementation is the default one and requires no configuration at all.
+ * `PooledExecutionService` Keeps a configurable set of thread pools and divides them to service all executor service
+   requests. This implementation must be configured with a `PooledExecutionServiceConfiguration` when needed.
+
+=== Configuring `PooledExecutionService`
+
+ * Done via `PooledExecutionServiceConfiguration`
+ * There can be at most one default pool, add it via `addDefaultPool`. Its name does not matter.
+ * There can be any number of pools. Add them via `addPool`.
+
+=== Using the configured thread pools
```

Comment:
This section needs to be fleshed out a bit better. Would be good to have more context on why you add class names here.